### ISSUE TITLE
Consider restricting the creation of sudt/xudt cells with empty amount

### ISSUE THREAD

**Issue by tea2x** [nervosnetwork/ckb-production-scripts#82]
There are xudt/sudt cells with zero balance being created. For example: This [CAT++ transaction](https://explorer.nervos.org/transaction/0xd6c7c03ca7f17b055b5118d407d3ed4663f9075620cb280b81b5ae78fc6d0c04), output 3 is an xudt cell with zero balance.
 
The creation of this type of cells either by mistakes or by intention are meaningless.
Restricting them can help keep CKB UTXO set minimal and lightweight.

**Comment by XuJiandong** [comment]
You're correct, these cells are meaningless as they can't be spent anymore. Note that these cells can be recycled by attaching them to a minting transaction.

**Comment by quake** [comment]
> Note that these cells can be recycled by attaching them to a minting transaction(normal transaction can't consume it).

can we combine a zero udt cell with a non-zero udt cell as inputs and transfer them to one output cell?

**Comment by tea2x** [comment]
>Note that these cells can be recycled by attaching them to a minting transaction(normal transaction can't consume it).

yeah now users have to do additional tasks to consolidate these cells but it's sort of cumbersome.

I understand your comment in https://github.com/nervosnetwork/ckb-production-scripts/pull/83#issuecomment-2079024979 that production scripts can't be upgraded but if this is really considered... can it be batched in the next hardfork?

**Comment by XuJiandong** [comment]
The hardfork cannot upgrade such a script. Instead, you can deploy a new one with a patch to the mainnet, and then perform a migration.

**Comment by tea2x** [comment]
Ok. Thanks for the confirmation.
